spec/formatting.md

```diff
@@ -230,65 +231,54 @@ the following steps are taken:
 - A _private-use_ _annotation_ is unsupported by the implementation or if
   a _private-use_ _annotation_ fails to resolve.
 - The _expression_ has a _reserved_ _annotation_.
 
 The _fallback value_ depends on the contents of the _expression_:
 
-- _expression_ with _literal_ _operand_: U+007C VERTICAL LINE `|`
-  followed by the value of the Literal,
-  and then by U+007C VERTICAL LINE `|`.
-  The same representation is used for both _quoted_ and _unquoted_ values.
-
-  > Examples: `|your horse|`, `|42|`
+- _expression_ with _annotation_ and no _operand_:
+  the _annotation_ starting sigil followed by its _identifier_
 
-- _expression_ with _variable_ _operand_: U+0024 DOLLAR SIGN `$`
-  followed by the _variable_ _name_ of the _operand_
+  > Examples:
+  > In a context where `:func` fails to resolve, `{:func}` resolves to the _fallback value_ `:func`.
+  > `{@reserved |...|}` always resolves to the _fallback value_ `@reserved`.
 
-  > Example: `$user`
+- _expression_ with _literal_ _operand_:
+  U+007C VERTICAL LINE `|`
+  followed by the value of the _literal_
+  with escaping applied to U+005C REVERSE SOLIDUS `\` and U+007C VERTICAL LINE `|`,
+  and then by U+007C VERTICAL LINE `|`.
+  The same representation is used for both _quoted_ and _unquoted_ values.
 
-- _expression_ with no _operand_:
-  the _function_ starting sigil followed by its _name_
+  > Examples:
+  > In a context where `:func` fails to resolve,
+  > `{42 :func}` resolves to the _fallback value_ `|42|` and
+  > `{|C:\\| :func}` resolves to the _fallback value_ `|C:\\|`.
+  > `{|| @reserved}` always resolves to the _fallback value_ `||`.
 
-  > Examples: `:platform`, `+tag`, `-tag`
+- _expression_ that consists of a _variable_, or _expression_ with _variable_ _operand_:
+  - If the _variable_ fails to resolve, or resolves to a value that is not a _fallback value_:
+    U+0024 DOLLAR SIGN `$` followed by the _name_ of the _variable_
 
-- Otherwise: The U+FFFD REPLACEMENT CHARACTER `�` character
+    > Examples:
+    > In a context where `$var` fails to resolve, `{$var}` and `{$var :number}` and `{$var @reserved}`
+    > all resolve to the _fallback value_ `$var`.
+    > In a context where `:func` fails to resolve, `.input $arg {{{$arg :func}}}` resolves to the _fallback value_ `$arg`.
 
-_Option_ names and values are not included in the _fallback value_.
+  - If the _variable_ resolves to a _fallback value_: the _fallback value_ to which it resolves
 
-When an error occurs in an _expression_ with a _variable_ _operand_
-and the _variable_ refers to a local _declaration_,
-the _fallback value_ is formatted based on the _expression_
-on the right-hand side of the _declaration_,
-rather than the _expression_ in the _selector_ or _pattern_.
+    > In a context where `:now` fails to resolve but `:datetime` does not,
+    > `.local $t = {:now format=iso8601} .local $pretty_t = {$t :datetime} {{{$pretty_t}}}` resolves to the _fallback value_ `:now`.
```

Comment:
Two comments:

1. This is tricky. If `$t` is evaluated eagerly (which we allow), then `$pretty_t` might contain a different sort of error (since `$t` contains a string that isn't a representative of a date/time value). Is this what we want here?

2. I find the examples hard to read vs. what we had before. I think the whitespace and multiline presentation of the original example (which is fine to delete) could be beneficial to replicate in the new examples.

Reply:
The second point is now handled, but the first seems like a consequence of logic we already have—I'm just highlighting its transitive nature:
1. `:now` fails to resolve, so `.local $t = {:now format=iso8601}` resolves to the fallback value ":now".
2. Therefore `.local $pretty_t = {$t :datetime}` resolves to the same fallback value ":now", even in the [current text](https://github.com/unicode-org/message-format-wg/blob/c0abd4f793d0a6f10807b5f8f2d44fe016860db2/spec/formatting.md#fallback-resolution)—"When an error occurs in an <em>expression</em> with a <em>variable</em> <em>operand</em> and the <em>variable</em> refers to a local <em>declaration</em>, the <em>fallback value</em> is **formatted based on the <em>expression</em> on the right-hand side of the <em>declaration</em>**…", followed by an explicit example, and also "If a <em>variable</em> would resolve to a <em>fallback value</em>,
this MUST also be considered a failure" text in [Variable Resolution](https://github.com/unicode-org/message-format-wg/blob/c0abd4f793d0a6f10807b5f8f2d44fe016860db2/spec/formatting.md#variable-resolution)
3. Therefore `{$pretty_t}` resolves to the same fallback value ":now" for the same reason.

Reply:
The main intent with the current transitive logic is to prevent the name of a local variable from ever showing up in the output, either during fallback or as the `source` of a formatted part.

Local variables should be considered message-internal implementation details that should never leak outside it.

Reply:
...which I believe is demonstrated in the new example.